src/utils.rs

```diff
@@ -1,21 +1,33 @@
-use chrono::{Utc, TimeZone};
+use chrono::{DateTime, TimeZone, Utc, Local, Duration};
 use md4::Md4;
 use md5::{Digest, Md5};
 use hmac::{Hmac, Mac};
 use bytes::{Bytes, Buf};
 
-pub fn get_windows_time() -> u64 {
-    let back_then = Utc.ymd(1601, 1, 1).and_hms(0, 1, 1);
-    let now = Utc::now();
+/// From MS-DYTP 2.2.3: FILETIME is a 64bit value, representing the
+/// number of 100-nanosecond intervals that have elapsed since January
+/// 1, 1601 in UTC.
```

Comment:
Since I regularly wonder why it's 1601 and not 1600; the reason, [as described here](https://devblogs.microsoft.com/oldnewthing/20090306-00/?p=18913):

> The Gregorian calendar operates on a 400-year cycle, and 1601 is the first year of the cycle that was active at the time Windows NT was being designed. In other words, it was chosen to make the math come out nicely. 

I assume that the 100-ns precision is not associated with a specific requirement but just the shortest base-10-divisible precision that would fit into 64bit.
And given that it's an unsigned int, I assume that in CIFS one cannot back-date ones files to before that time :zany_face:

Reply:
Thx for the link, i did'nt know :-) I don't know why they choose 10th of a microsecond as unit.. nanoseconds would have worked too (for around 585 years, counting from 1601, if i'm not mistaken). Maybe 10th of ms just felt safer to them.